### PR TITLE
[PluggableDevice] Allow quantized type registration.

### PR DESCRIPTION
**Pull request by quintinwang5**
Support quantized types for PluggableDevice's kernel.

### CODE REVIEW

**Comment by quintinwang5** [comment]
@penpornk @annarev  This PR is for the quantized data type support for plugin, please help to review. Thanks.

File: tensorflow/c/kernels_test.cc [hunk 1]
```diff
@@ -589,50 +589,77 @@ TEST(TestKernel, DeleteKernelBuilderIsOkOnNull) {
...
-  TF_KernelBuilder_TypeConstraint(builder, "T", TF_DataType::TF_INT32, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-  TF_RegisterKernelBuilder(node_name, builder, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-
-  TF_Buffer* buf = TF_GetRegisteredKernelsForOp(op_name, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-  KernelList list;
-  list.ParseFromArray(buf->data, buf->length);
+std::string expected_string(const char* type) {
```
Comment (annarev) [inline tensorflow/c/kernels_test.cc:592]: nit. Use camel case for function name: `ExpectedString`

File: tensorflow/c/kernels_test.cc [hunk 2]
```diff
@@ -589,50 +589,77 @@ TEST(TestKernel, DeleteKernelBuilderIsOkOnNull) {
...
+                            std::string("\"") +
+                            R"str(
   device_type: "FakeDeviceName1"
   constraint {
     name: "T"
     allowed_values {
       list {
-        type: DT_INT32
+        type: )str" + std::string(type) +
+                            R"##(
```
Comment (annarev) [inline tensorflow/c/kernels_test.cc:603]: May be use `StrFormat` instead to make it easier to read. Example: https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/stream_executor/gpu/redzone_allocator.cc;l=69;drc=4648d99eaecddd47a1fd5209a620635733726492

Comment (quintinwang5) [reply tensorflow/c/kernels_test.cc]: Done. And raw string is necessary here for comparing with protobuf generated string.

**Comment by jzhoulon** [comment]
@penpornk This PR is related with kernel C API, could you help to review or suggest other reviewers? Thanks very much.

**Comment by penpornk** [comment]
@saxenasaurabh Could you please help suggest a reviewer for this PR as well? Thank you very much!

File: tensorflow/c/kernels.cc [hunk 3]
```diff
@@ -81,6 +81,9 @@ void AddTypeConstraint(TF_KernelBuilder* kernel_builder, const char* attr_name,
   // TF_CALL_ALL_TYPES macro can find tensorflow::string as string.
   switch (dtype) {
     TF_CALL_ALL_TYPES(CASE);
+    TF_CALL_QUANTIZED_TYPES(CASE);
+    TF_CALL_quint16(CASE);
```
Comment (saxenasaurabh) [inline tensorflow/c/kernels.cc:85]: @jianlijianli do you know why we do not include quint16 and qint16 in `TF_CALL_QUANTIZED_TYPES`?

Comment (jianlijianli) [reply tensorflow/c/kernels.cc:85]: That macro was added quite a while ago and I don't have much context but it seems quint16 and qint16 was not needed in the original use case but there is a [TODO](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/register_types.h#L205-L207) to include those types.

Comment (saxenasaurabh) [reply tensorflow/c/kernels.cc:85]: Ok, thanks for clarifying. I guess this could be cleaned up separately.

**Review by saxenasaurabh — APPROVED** [review]

**Pull request merged**